hsdio.py

```diff
@@ -32,111 +33,83 @@ class HSDIO(Instrument): # could inherit from an Instrument class if helpful
     dllpath32 = os.path.join(f"C:\{programsDir32}\IVI Foundation\IVI\Bin", "niHSDIO.dll")
     dllpath64 = os.path.join("C:\Program Files\IVI Foundation\IVI\Bin", "niHSDIO_64.dll")
 
-    def __init__(self, pxi):
-        super().__init__(pxi, "HSDIO")
-
-        ## device settings
-        self.logger = logging.getLogger(str(self.__class__))
-        self.pxi = pxi
+    def __init__(self, pxi, node: ET.Element = None):
+        # device settings
         self.resourceNames = np.array([], dtype=str)
-        self.clockRate = 2*10**7 # 20 MHz
+        self.clockRate = 2*10**7  # 20 MHz
         self.hardwareAlignmentQuantum = 1 # (in samples)
         self.activeChannels = np.array([], dtype=c_int32)
         self.initialStates = np.array([], dtype=str)
         self.idleStates = np.array([], dtype=str)
         self.pulseGenScript = """script script 1
                                       wait 1
                                    end script"""
-        self.scriptTriggers = []
-        self.startTrigger = StartTrigger()
-
+        self.scriptTriggers: List[Trigger] = []
+        self.startTrigger: StartTrigger = StartTrigger()
+        self.description = ""
 
-        # These two have are related to one another, each session is attached to a handle, each handle can support man
-        # sessions. Sessions now have an attribute HsdioSession.handle (a python string)
-        self.instrumentHandles = []  # array to hold instrument handles; CAN PROBABLY DELETE; sessions takes care of this
-        self.sessions = []  # array to hold HsdioSession objects
-        self.waveformArr = []
+        self.sessions: List[HSDIOSession] = []
+        self.waveformArr: List[HSDIOWaveform] = []
 
         # whether or not we've actually populated the attributes above
         self.isInitialized = False
-        
-        
+        # check this to see if waveform has been written and updated without error
+        self.wvf_written = False
+
+        super().__init__(pxi, "HSDIO", node)
+
     def load_xml(self, node):
         """
         iterate through node's children and parse xml by tag to update HSDIO
         device settings
         'node': type is ET.Element. tag should be "HSDIO"
         """
 
-        assert node.tag == self.expectedRoot, "This XML is not tagged for the HSDIO"
```

Comment:
Not sure how the alternate option is different from the super call but I do agree that it's not super transparent.

I just wanted to remove code duplication and this seemed like the way to go since we already have the parent class set up.

Reply:
I agree with the spirit of the change for sure. Maybe leave the super call but put a comment above it, e.g. "verify the xml is for this device, etc".